### PR TITLE
[PyCDE] Add support for placing CompRegOp.

### PR DESCRIPTION
**Pull request by mikeurbach**
This extends the Instance wrapper to wrap not just instances but also
registers. Additionally, in the Tcl export, this looks for registers,
adds them to the symbol cache, and uses their symbol names.

Users are now able to place CompRegOp when its innerSym attribute is
set. If they create a CompRegOp directly, the inner_sym must be passed
explicitly, but the `reg(...)` helper on Value is extended to use the
name as the inner_sym when passed.

### CODE REVIEW

File: frontends/PyCDE/test/compreg.py [hunk 1]
```diff
@@ -19,16 +22,26 @@ class CompReg:
...
-    compreg = seq.CompRegOp(types.i8, clk=ports.clk, input=ports.input)
+    compreg = seq.CompRegOp(types.i8,
+                            clk=ports.clk,
+                            input=ports.input,
+                            name="reg")
     ports.output = compreg
 
 
+appid = pycde.AppIDIndex()
+loc = placement([], PrimitiveType.M20K, 0, 0, 0)
```
Comment (teqdruid) [inline frontends/PyCDE/test/compreg.py:33]: Plz add a "reg" type to PrimitiveType.

**Review by teqdruid — APPROVED** [review]
Does Export tcl support this? If your test passes, then yeah. So this is totally the way to go.

File: frontends/PyCDE/src/pycde/value.py [hunk 2]
```diff
@@ -52,7 +52,8 @@ def reg(self, clk, rst=None, name=None):
                            input=self.value,
                            clk=clk,
                            reset=rst,
-                           name=name)
+                           name=name,
+                           inner_sym=name)
```
Comment (mikeurbach) [inline frontends/PyCDE/src/pycde/value.py:56]: @teqdruid following the discussion in https://github.com/llvm/circt/pull/2303#pullrequestreview-826959487, I took the liberty of setting the inner_sym to name when it is provided here. I think this is what our users would expect, but let me know if you disagree. If they create the CompRegOp themselves, they still have the ability to have full control over name versus symbol.

**Review by mikeurbach — COMMENTED** [review]
@teqdruid please take another look when you get a chance. The PyCDE change is roughly the same, but uses the new innerSym attribute rather than overloading the meaning of the name attribute. A similar change is made in ExportQuartusTcl now as well, since the generic logic to fall back to looking for a name attribute no longer works for this case.

File: frontends/PyCDE/test/compreg.py [hunk 3]
```diff
@@ -19,16 +22,26 @@ class CompReg:
...
 mod.generate()
+mod.get_instance(CompReg).walk(appid.apply_attributes_visitor)
 mod.print()
 mod.emit_outputs()
 
 # CHECK: reg [7:0] [[NAME:.+]];
 # CHECK: always_ff @(posedge clk)
 # CHECK: [[NAME]] <= {{.+}}
+
+# TCL: set_location_assignment FF_X0_Y0_N0 -to $parent|{{.+}}
```
Comment (teqdruid) [inline frontends/PyCDE/test/compreg.py:47]: Could you specify the full path here?

**Review by teqdruid — APPROVED** [review]
Thanks!

**Pull request merged**